From ef115a0dcbbb4b771e4a9051080b8bba53ae96e2 Mon Sep 17 00:00:00 2001
From: Bert van Hall <bert.vanhall@avionic-design.de>
Date: Wed, 7 Dec 2016 10:56:24 +0100
Subject: [PATCH] fix openssl 1.1.x detection

The SSL_library_init function has been renamed to OPENSSL_init_ssl from
openssl 1.1.0 on. While the old name still exists as a define for
backwards compatibility, this breaks detection in the library itself.
Update configure.ac to just detect the library instead of specific
functions.

Signed-off-by: Bert van Hall <bert.vanhall@avionic-design.de>
---
 configure.ac | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 1b73966..6b1cf8e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -125,12 +125,11 @@ fi
 AH_TEMPLATE(HAVE_X509_PRINT_EX_FP, [open ssl X509_print_ex_fp available])
 if test "x$with_ssl" != "xno"; then
 	if test "x$HAVE_LIBCRYPTO" = "xtrue"; then
-		AC_CHECK_LIB(ssl, SSL_library_init,
+		PKG_CHECK_MODULES(OPENSSL, [openssl >= 1.0.0],
 			SSL_LIBS="-lssl -lcrypto"
-			[AC_DEFINE(HAVE_LIBSSL) HAVE_LIBSSL="true"], ,
-			-lcrypto)
+			[AC_DEFINE(HAVE_LIBSSL) HAVE_LIBSSL="true"], ,)
 	else
-		AC_CHECK_LIB(ssl, SSL_library_init,
+		PKG_CHECK_MODULES(OPENSSL, [openssl >= 1.0.0],
 			SSL_LIBS="-lssl"
 			[AC_DEFINE(HAVE_LIBSSL) HAVE_LIBSSL="true"], ,)
 	fi
-- 
2.7.4

